lib/reek/core/smell_repository.rb

```diff
@@ -24,7 +24,9 @@ def initialize(source_description = nil, smell_types = self.class.smell_types)
       end
 
       def configure(klass, config)
-        @detectors[klass].configure_with(config) if @detectors[klass]
+        detector = @detectors[klass]
+        raise ArgumentError, "Cannot configure smell #{klass}" unless detector
```

Comment:
"Cannot configure smell ..." is a little confusing for the user since it tells them basically "I have a valid detector but for whatever reasons I can't configure it" when we actually want to say "you give us a smell detector we don't know".

How about changing this to

>> "Cannot configure nknown smell detector #{klass}"